packages/front-end/components/Settings/EditDataSourceSettingsForm.tsx

```diff
@@ -33,6 +34,9 @@ const EditDataSourceSettingsForm: FC<{
   }, [source]);
 
   const { apiCall } = useAuth();
+  const [addNameCols, setNameCols] = useState(
+    data.settings?.queries?.exposure[0]?.hasNameCol
+  );
```

Comment:
This should not be a single global state.  It's a separate setting for each exposure query, so you can just use the form value directly.

Reply:
Got it. I guess I just figured that we could assume that if one exposure query object had the name columns, then it would make sense for all of them to have it.

Reply:
Resolved by new changes